core/trino-main/src/main/java/io/trino/sql/planner/NodePartitioningManager.java

```diff
@@ -171,18 +170,14 @@ public NodePartitionMap getNodePartitioningMap(Session session, PartitioningHand
         return new NodePartitionMap(partitionToNode, bucketToPartition, getSplitToBucket(session, partitioningHandle));
     }
 
-    public BucketNodeMap getBucketNodeMap(Session session, PartitioningHandle partitioningHandle, boolean preferDynamic)
+    public BucketNodeMap getBucketNodeMap(Session session, PartitioningHandle partitioningHandle)
     {
         ConnectorBucketNodeMap connectorBucketNodeMap = getConnectorBucketNodeMap(session, partitioningHandle);
 
         if (connectorBucketNodeMap.hasFixedMapping()) {
             return new FixedBucketNodeMap(getSplitToBucket(session, partitioningHandle), getFixedMapping(connectorBucketNodeMap));
         }
 
-        if (preferDynamic) {
```

Comment:
Can you do that? There are calls with `preferDynamic` set to `true` and `false`

Reply:
I am afraid you are right. I dropped the last commit.